packages/core/test/reconciliation/futures/reconcileNamedEncodeFunctionCall.ts

```diff
@@ -0,0 +1,376 @@
+          status: ExecutionStatus.STARTED,
+          functionName: "function1",
+        }
+      )
+    );
+
+    assert.deepStrictEqual(reconiliationResult.reconciliationFailures, [
+      {
+        futureId: "Module#config",
+        failure:
+          "A dependency from Module#config to Module#Contract1 has been added. The former has started executing before the latter started executing, so this change is incompatible.",
+      },
+    ]);
+  });
+
+  it("should find changes to function name unreconciliable", async () => {
+    const moduleDefinition = buildModule("Module", (m) => {
+      const contract1 = m.contract("Contract1");
+
+      m.encodeFunctionCall(contract1, "functionChanged", [], { id: "config" });
+
+      return { contract1 };
+    });
+
+    const reconiliationResult = await reconcile(
+      moduleDefinition,
+      createDeploymentState(
+        {
+          ...exampleDeploymentState,
+          id: "Module#Contract1",
+          status: ExecutionStatus.SUCCESS,
+          result: {
+            type: ExecutionResultType.SUCCESS,
+            address: differentAddress,
+          },
+        },
+        {
+          ...exampleEncodeFunctionCallState,
+          id: "Module#config",
+          futureType: FutureType.ENCODE_FUNCTION_CALL,
+          status: ExecutionStatus.STARTED,
+          functionName: "functionUnchanged",
+        }
+      )
+    );
+
+    assert.deepStrictEqual(reconiliationResult.reconciliationFailures, [
+      {
+        futureId: "Module#config",
+        failure:
+          "Function name has been changed from functionUnchanged to functionChanged",
+      },
+    ]);
+  });
+
+  it("should find changes to function args unreconciliable", async () => {
+    const moduleDefinition = buildModule("Module", (m) => {
+      const ticker = m.getParameter("ticker", "CHANGED");
+
+      const contract1 = m.contract("Contract1");
+
+      m.encodeFunctionCall(contract1, "function1", [[ticker]], {});
+
+      return { contract1 };
+    });
+
+    const reconiliationResult = await reconcile(
+      moduleDefinition,
+      createDeploymentState(
+        {
+          ...exampleDeploymentState,
+          id: "Module#Contract1",
+          status: ExecutionStatus.SUCCESS,
+          result: {
+            type: ExecutionResultType.SUCCESS,
+            address: differentAddress,
+          },
+        },
+        {
+          ...exampleEncodeFunctionCallState,
+          id: "Module#Contract1.function1",
+          futureType: FutureType.ENCODE_FUNCTION_CALL,
+          status: ExecutionStatus.STARTED,
+          functionName: "function1",
+          args: [["UNCHANGED"]],
+        }
+      )
+    );
+
+    assert.deepStrictEqual(reconiliationResult.reconciliationFailures, [
+      {
+        futureId: "Module#Contract1.function1",
+        failure: "Argument at index 0 has been changed",
+      },
+    ]);
+  });
+
+  it("should reconcile an address arg with entirely different casing", async () => {
```

Comment:
Lets drop this test. I suspect it is pulled over, but I don't think it adds here. Same for the next test.